libs/aws/langchain_aws/rerank/rerank.py

```diff
@@ -0,0 +1,142 @@
+from copy import deepcopy
+from typing import Any, Dict, List, Optional, Sequence, Union
+
+import boto3
+from langchain_core.callbacks.manager import Callbacks
+from langchain_core.documents import BaseDocumentCompressor, Document
+from langchain_core.utils import from_env
+from pydantic import ConfigDict, Field, model_validator
+from typing_extensions import Self
+
+
+class BedrockRerank(BaseDocumentCompressor):
+    """Document compressor that uses AWS Bedrock Rerank API."""
+
+    client: Any = None
+    """Bedrock client to use for compressing documents."""
+    top_n: Optional[int] = 3
+    """Number of documents to return."""
+    model_id: Optional[str] = "amazon.rerank-v1:0"
+    """Model ID to fetch ARN dynamically."""
+    aws_region: str = Field(
+        default_factory=from_env("AWS_DEFAULT_REGION", default=None)
+    )
+    """AWS region to initialize the Bedrock client."""
+    aws_profile: Optional[str] = Field(
+        default_factory=from_env("AWS_PROFILE", default=None)
+    )
```

Comment:
Similar to the region, would like this to be consistent with other implementations in langchain_aws.

```suggestion
    credentials_profile_name: Optional[str] = Field(
        default_factory=from_env("AWS_PROFILE", default=None)
    )
```

Reply:
Solved on 04891259cee476800dfe2de507faefc61ada9115